moveit_core/robot_model/include/moveit/robot_model/link_model.h

```diff
@@ -41,13 +41,16 @@
 #include <vector>
 #include <utility>
 #include <map>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 #include <moveit/macros/class_forward.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
I was assuming we were not switching these because it would violate the ROS policy of no std::shared_ptrs in the public headers of packages in ros-desktop-full. But then again, we already had to break that rule for FCL and Octomap, so it doesn't make a difference going all the way now.